crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -292,7 +292,7 @@ impl BaseClient {
 
         trace!("ready to submit changes to store");
         store.save_changes(&changes).await?;
-        self.apply_changes(&changes, false);
+        self.apply_changes(&changes, true);
```

Comment:
The room stream won't fire if the room already exists. A room info update is likely to trigger a room list update if the latest event changes for example. Hence the `true` here. I'm adding a comment.

Reply:
I think this would lead to spurious updates, because we're already saving room infos explicitly here. (The room state management is a bit of a mess, we should likely add this to the spring cleaning list). Putting `true` will force a spurious update here, which might not be an issue in testing, so let's see if we can keep it.

Reply:
Ideally, we want to detect whether an update is necessary instead of always coercing to `true`. But let's do that in another PR.

Reply:
The way it worked before: changing the room's data meant the vectordiff would include a change for that room (even if it's not moved position in the list: just to notify that there's a change in the room, so the subscribers can be notified). Hence, we didn't need to pass `true` here, since the change would happen automatically.

Has this changed? Is this the actual reason why you need to pass `true` here?